### PR TITLE
feat: add workflows/README.md to elaborate on GitHub workflows

### PR DESCRIPTION
**Pull request by civsiv**
Part of https://github.com/openactive/OpenActive.Server.NET/issues/211

### CODE REVIEW

**Review by lukehesluke — COMMENTED** [review]
This is really thorough documentation @civsiv !

One thing I'm wondering about is the coupling between the documentation and the code. This is always an issue for sure, but I think in this case, as the documentation here is so methodical, small changes to the logic would require changes to the README file (or otherwise the README docs will become stale).
Additionally, some of the docs are quite "low-level" (for want of a better phrase) e.g. descriptions of each step, and might therefore be more useful in the code, for someone who needs to make a change or is debugging the logic.

So my suggestion is:

* Migrate much of this to code comments, where the description of a step can be right next to the step itself.
* Instead have the README.md file include only summary-level information, like what each job does and what each workflow does
* Have the README.md make clear that more specific details can be found in code comments in the .yml files

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+### publish-server job
+This job publishes the OpenActive.Server.NET project to Nuget. It then creates a release in the GitHub repository.
+
+
+### publish-fake-database job
+Very similar to the above `publish-server` job, this job publishes the Fake Database project within OpenActive.Server.NET to Nuget.
+
+## create-dependencies-pr.yaml
+
+This workflow updates OpenActive.Server.NET's OA dependencies if there has been a change in one of the projects. This therefore keeps Server.NET up-to-date with the rest of the OA ecosystem.
```
Comment (lukehesluke) [inline .github/workflows/README.md:42]:
```suggestion
This workflow creates a pull request to OpenActive.Server.NET's OpenActive (OA) dependencies if there has been a change in one of those projects. This therefore keeps Server.NET up-to-date with the rest of the OA ecosystem.
```

File: .github/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+This workflow can be triggered in two ways:
+
+- **workflow_dispatch**: This allows you to manually trigger the workflow through the GitHub Actions UI.
+
+- **repository_dispatch**: This allows the workflow to be triggered when a custom repository dispatch event is sent. It listens for dispatch events from two OA repos, specifically when they are updated: `OpenActive.NET-update` and `OpenActive.DatasetSite.NET-update`.
```
Comment (lukehesluke) [inline .github/workflows/README.md:48]: This is probably just because of my lack of knowledge, but I don't understand this

Comment (lukehesluke) [inline .github/workflows/README.md:48]:
```suggestion
- **repository_dispatch**: This allows the workflow to be triggered when a custom repository dispatch event is sent. It listens for dispatch events from two OA repos, specifically, when they are updated: `OpenActive.NET-update` and `OpenActive.DatasetSite.NET-update`. 
```

(i think)

**Review by lukehesluke — APPROVED** [review]

File: .github/workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+### core job
+This job builds and runs the `OpenActive.Server.NET` project. Then it checks out Test Suite and runs the matrix of tests the `OpenActive.Server.NET` project.
+
+### framework job
+This job is very similar to the above but builds `OpenActive.Server.NET` using .NET Framework and runs Test Suite against it.
+
+
+### deploy-reference-implementation job
+This job deploys the `master` branch of Reference Implementation Booking Server and Identity Server Reference Implementation to two Azure webapps
```
[the remark this answers is not in the document]
Comment (civsiv) [reply .github/workflows/README.md:30]: Not really, but will link to where the poor docs are

File: .github/workflows/README.md (hunk 2 above)
Comment (civsiv) [reply .github/workflows/README.md:48]: I've reworded and added link to docs

**Pull request merged**